mne/label.py

```diff
@@ -1659,6 +1660,162 @@ def _grow_nonoverlapping_labels(subject, seeds_, extents_, hemis, vertices_,
     return labels
 
 
+def random_parcellation(subject, n_parcel, hemi, subjects_dir=None,
+                        surface='white', random_state=None):
+    """Generate random cortex parcellation by growing labels.
+
+    This function generates a number of labels which don't intersect and
+    cover the whole surface. Regions are growing around randomly chosen
+    seeds.
+
+    Parameters
+    ----------
+    subject : string
+        Name of the subject as in SUBJECTS_DIR.
+    n_parcel : int
+        Total number of cortical parcels.
+    hemi : str
+        hemisphere id (ie 'lh', 'rh', 'both'). In the case
+        of 'both', both hemispheres are processed with (n_parcel // 2)
+        parcels per hemisphere.
+    subjects_dir : string
+        Path to SUBJECTS_DIR if not set in the environment.
+    surface : string
+        The surface used to grow the labels, defaults to the white surface.
+    random_state : None | int | np.random.RandomState
+        To specify the random generator state.
+
+    Returns
+    -------
+    labels : list of Label
+        Random cortex parcellation
+    """
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    if hemi == 'both':
+        hemi = ['lh', 'rh']
+    hemis = np.atleast_1d(hemi)
+
+    # load the surfaces and create the distance graphs
+    tris, vert, dist = {}, {}, {}
+    for hemi in set(hemis):
+        surf_fname = op.join(subjects_dir, subject, 'surf', hemi + '.' +
+                             surface)
+        vert[hemi], tris[hemi] = read_surface(surf_fname)
+        dist[hemi] = mesh_dist(tris[hemi], vert[hemi])
+
+    # create the patches
+    labels = _cortex_parcellation(subject, n_parcel, hemis, vert, dist,
+                                  random_state)
+
+    # add a unique color to each label
+    colors = _n_colors(len(labels))
+    for label, color in zip(labels, colors):
+        label.color = color
+
+    return labels
+
+
+def _cortex_parcellation(subject, n_parcel, hemis, vertices_, graphs,
+                         random_state=None):
+    """Random cortex parcellation."""
+    labels = []
+    for hemi in set(hemis):
+        parcel_size = len(hemis) * len(vertices_[hemi]) // n_parcel
+        graph = graphs[hemi]  # distance graph
+        n_vertices = len(vertices_[hemi])
+
+        # prepare parcellation
+        parc = np.full(n_vertices, -1, dtype='int32')
+
+        # initialize active sources
+        rng = check_random_state(random_state)
```

Comment:
I would have put the check_random_state outside of the for loop on hemi no?

Reply:
Yes, done.